src/components/battery/BatteryController.cpp

```diff
@@ -1,49 +1,89 @@
 #include "BatteryController.h"
 #include <hal/nrf_gpio.h>
+#include <nrfx_saadc.h>
 #include <libraries/log/nrf_log.h>
 #include <algorithm>
+#include <math.h>
 
 using namespace Pinetime::Controllers;
 
+#define SAMPLES_IN_BUFFER 1
```

Comment:
These constants and static variables can be move into the Battery class, as member variables.

Reply:
Ok, now I understand why you declared them as static variables instead of member variables... Too bad that function pointer does not provide a 'user data' like lvgl to avoid these statics or this ::Instance() method!

Reply:
yes the sdk functions have this problem.
but I still rewrote it and now it's better.